check-approved-requests.py

```diff
@@ -201,13 +201,16 @@ def check_requests(request_type, auth_file, worksheet_key):
                             **request_info)
  
             processed_rows.append(idx)
-            timestamp_spreadsheet(sheet, timestamp, processed_rows)
             if args.log:
                 log_request(args.log, timestamp, request_info['user_email'])
         else:
             # skip over unapproved or already-notified rows
             continue
 
+    # Google API returns an error if you send an empty request
+    if processed_rows:
```

Comment:
There are 2 reasons I went with if instead of try/except:

1. Sending requests to the Google API is kind of slow, so doing it needlessly makes the script take longer.  Usually it runs in the background but sometimes we run it manually.  It's better if it just exits right away when it doesn't have anything to do.  

2. The error returned is Http 404, so we would have to parse the error message to make sure we were only ignoring the specific error we wanted.  It seems silly to do that if we already knew that's what we would get.